bundles-lib/src/main/java/org/apache/metron/bundles/BundleSystem.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.metron.bundles.bundle.Bundle;
+import org.apache.metron.bundles.util.BundleProperties;
+import org.apache.metron.bundles.util.FileSystemManagerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * High level interface to the Bundle System.  While you may want to use the lower level classes it
+ * is not required, as BundleSystem provides the base required interface for initializing the system
+ * and instantiating classes
+ */
+public class BundleSystem {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  /**
+   * Builder for a BundleSystem. only {@link BundleProperties} are required. Beyond that, the
+   * BundleProperties, if they are the only parameter must have archive extension and bundle
+   * extension types properties present.
+   */
+  public static class Builder {
+
+    private BundleProperties properties;
+    private FileSystemManager fileSystemManager;
+    private List<Class> extensionClasses = new LinkedList<>();
+    private Bundle systemBundle;
+
+    /**
+     * The BundleProperties to use.  Unless other builder parameters override options
+     * (withExtensionClasses ), they must have archive extension and bundle extensions types
+     * specified
+     *
+     * @param properties The BundleProperties
+     * @return Builder
+     */
+    public Builder withBundleProperties(BundleProperties properties) {
+      this.properties = properties;
+      return this;
+    }
+
+    /**
+     * Provide a {@link FileSystemManager} to overide the default
+     *
+     * @param fileSystemManager override
+     * @return Builder
+     */
+    public Builder withFileSystemManager(FileSystemManager fileSystemManager) {
+      this.fileSystemManager = fileSystemManager;
+      return this;
+    }
+
+    /**
+     * Provide Extension Classes.  If not provided with this override then the classes will be
+     * configured from the BundleProperties. If provided, the properties file will not be used.
+     *
+     * @param extensionClasses override
+     * @return Builder
+     */
+    public Builder withExtensionClasses(List<Class> extensionClasses) {
+      this.extensionClasses.addAll(extensionClasses);
+      return this;
+    }
+
+    /**
+     * Provide a SystemBundle.  If not provided with this override then the default SystemBundle
+     * will be created.
+     */
+    public Builder withSystemBundle(Bundle systemBundle) {
+      this.systemBundle = systemBundle;
+      return this;
+    }
+
+    /**
+     * Builds a new BundleSystem.
+     *
+     * @return BundleSystem
+     * @throws NotInitializedException if any errors happen during build
+     */
+    public BundleSystem build() throws NotInitializedException {
+      if (this.properties == null) {
+        throw new IllegalArgumentException("BundleProperties are required");
```

Comment:
I will make it more clear they will not be used for the classes, but will be used for other things